src/poetry/core/factory.py

```diff
@@ -464,11 +464,20 @@ def validate(
                         continue
 
                     extras = script.get("extras", [])
+                    if extras:
+                        result["warnings"].append(
+                            f'The script "{name}" depends on an extra. Scripts'
+                            " depending on extras are deprecated and support for them"
+                            " might be removed in a future version of"
```

Comment:
```suggestion
                            " will be removed in a future version of"
```
Let's avoid "maybe"s in deprecations. Especially since this will be changed with PEP 621 support.